MovieApp/PresentationLayer/Components/SearchBar/SearchBar+Design.swift

```diff
@@ -0,0 +1,93 @@
+import UIKit
+
+extension SearchBar: ConstructViewsProtocol {
+
+    func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    func createViews() {
+        searchField = BaseSearchField()
+        searchField.addTarget(self, action: #selector(showCancelButton), for: .editingDidBegin)
+
+        let searchIcon = UIImageView(image: UIImage(named: "Search"))
+        searchIcon.contentMode = .center
+        searchField.leftView = searchIcon
+
+        let deleteButton = UIButton()
+        deleteButton.setImage(UIImage(named: "X"), for: .normal)
+        deleteButton.contentMode = .center
+        deleteButton.addTarget(self, action: #selector(clearSearchBar), for: .touchUpInside)
+        searchField.rightView = deleteButton
+        deleteButton.isHidden = true
+        addArrangedSubview(searchField)
+
+        cancelButton = UIButton()
+        cancelButton.addTarget(self, action: #selector(hideCancelButton), for: .touchUpInside)
+        addArrangedSubview(cancelButton)
+    }
+
+    func styleViews() {
+        searchField.backgroundColor = UIColor(named: "Gray")
+        searchField.layer.cornerRadius = 10
+        searchField.leftViewMode = .always
+        searchField.rightViewMode = .always
+
+        let placeholder = NSAttributedString(
+            string: "Search",
+            attributes: [
```

Comment:
To make this array more clear, extract fonts into variables. Also, do you need to cast to Any?

Reply:
The problem is that the font is an optional when constructed like this.
If I don't cast to Any XCode gives me a warning saying: Expression implicitly coerced from 'UIFont?' to 'Any'.
I have the option to provide a default value, force unwrap or cast to Any.